samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java

```diff
@@ -211,9 +211,12 @@ private void validateConfig() {
           }
         }
       }
+
+      log.info("Inferred the partition count for inputs to the join operator {} as {}.", join.getOpId(), partitions);
```

Comment:
Can we move this to line 204 and include the name of the edge that caused this count to be inferred?